utils/components.py

```diff
@@ -61,3 +69,311 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+            ############################################## ERROR #######################################################
+            # Secret {secret_name} does not exist. Ensure you created a secret with this name in namespace {namespace} #
+            ############################################################################################################
+            """)
+            sys.exit(1)
+
+    request_auth = {"Authorization": f"Bearer {model_api_key}"}
+    request_body = {
+        "model": model_name,
+        "messages": [{"role": "user", "content": "tell me a funny joke."}],
+    }
+    resp = requests.post(
+        f"{model_endpoint}/chat/completions",
+        headers=request_auth,
+        data=json.dumps(request_body),
+        verify=os.environ["SDG_CA_CERT_PATH"],
+    )
+    if resp.status_code != 200:
+        print(f"""
+        ############################################ ERROR ####################################################
+        # Model Server {model_name} is unavailable. Ensure the model is up and it is ready to serve requests. #
+        #######################################################################################################
+        """)
+        sys.exit(1)
+    else:
+        print(f"""
+        ################### INFO #######################
+        # Model Server {model_name} is up and running. #
+        ################################################
+        """)
+
+
+@dsl.component(
+    base_image=PYTHON_IMAGE,
+    packages_to_install=["model-registry==0.2.13", "kserve==0.13"],
+)
+def test_model_registry(model_registry_endpoint: str):
+    from model_registry import ModelRegistry
+
+    registry = ModelRegistry(
+        server_address=model_registry_endpoint,
+        author="ilab pipeline",
+        is_secure=False,
+        user_token=open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/token"
+        ).readline(),
+    )
+    # This line should enforce pipeline to hit MR endpoint
+    if len(registry.get_registered_models().page_size(1)._next_page()) >= 0:
+        print(f"""
+        ########### INFO ##############
+        # Model Registry is available #
+        ###############################
+        """)
+
+
+@dsl.component(base_image=PYTHON_IMAGE)
+def test_training_operator():
+    import os
+    import sys
+
+    from kubernetes import client
+    from kubernetes.client.rest import ApiException
+
+    sa_path = "/var/run/secrets/kubernetes.io/serviceaccount"
+    configuration = client.Configuration()
+    configuration.api_key["authorization"] = open(f"{sa_path}/token").readline()
+    configuration.api_key_prefix["authorization"] = "Bearer"
+    configuration.host = f"https://{os.environ['KUBERNETES_SERVICE_HOST']}"
+    configuration.ssl_ca_cert = f"{sa_path}/ca.crt"
+    namespace = open(f"{sa_path}/namespace").readline()
+
+    with client.ApiClient(configuration) as api_client:
+        api_instance = client.CustomObjectsApi(api_client)
```

Comment:
Could you directly check if the CRD object is available rather check if there is a CR?

Reply:
I can't find a better way to check directly if CRD is present. If you have any ideas how to do that, I can make the change.

Reply:
Sorry, what I meant was the equivalent of `kubectl get crd pytorchjobs.kubeflow.org`.

Reply:
I just realized that the pipeline-runner service account won't have access to query the CRD, so let's just use `list_namespaced_custom_object` with the pod's current namespace.